src/main/java/com/fasterxml/jackson/databind/ser/BasicSerializerFactory.java

```diff
@@ -1243,7 +1243,10 @@ private void _removeEnumSelfReferences(BasicBeanDescription beanDesc) {
             BeanPropertyDefinition property = it.next();
             JavaType propType = property.getPrimaryType();
             // is the property a self-reference?
-            if (propType.isEnumType() && propType.isTypeOrSubTypeOf(aClass)) {
+            if (propType.isEnumType() && propType.isTypeOrSubTypeOf(aClass)
+                    // [databind#4564] Since 2.16.3, Enum's should allow self as field, so let's remove only if static.
+                    && property.getAccessor().isStatic())
```

Comment:
How is this even possible? Neither static Methods nor Fields should become or be part of POJO properties?
Except... was there some special handling that did just that, just for Enums?
So what is left to remove? I guess I can play with the test case and see what it sees...

Reply:
It's this check in `AnnotatedFieldCollector` here....

https://github.com/FasterXML/jackson-databind/blob/fc67817a0525d123b2f600c356ce5675a1da5b4c/src/main/java/com/fasterxml/jackson/databind/introspect/AnnotatedFieldCollector.java#L128

...  where we collect in `Enum`s then later at this point (current line) to remove self-references and such.

Reply:
Thus current method here is `_removeEnumSelfReferences()`, except having self as enum was not accounted for, up until now.